jboss/container/java/certs/bash/configure.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+# Configure module
+set -e
+
+SCRIPT_DIR=$(dirname $0)
+ARTIFACTS_DIR=${SCRIPT_DIR}/artifacts
+
+# necessary for random UID user to run update-ca-certs
```

Comment:
Is it possible to chmod these back afterwards? it seems odd to leave them 777.

Reply:
No; because the chmod is happening in the build stage but the write access (via update-ca-trust) is happening at runtime.

It's

Reply:
checking

Reply:
Yes g+w is sufficient. The threat model doesn't really change (same user in both cases) but it feels nicer to not have it as wide as 777. I'll commit this.

Reply:
Do we need to update the permissions if we're creating the link during image build?  Also, can the pod user run the update script without special privs?  Will the update script fail if the link is broken (i.e. the target file doesn't exist)?  Do we need to chmod the link, to make sure the perms are correct there (I had this problem in other places, where the link was root:root, which required an explicit `chown -h`.

Reply:
If all of the above works out okay (i.e. link has correct owner, runtime user can invoke the update script, not having to update perms on the directories), then I think that eliminates and problems.

Reply:
> Do we need to update the permissions if we're creating the link during image build?

Yes, because we are widening the permissions on the locations to which the derivative TLS bundles are written to. The "update-ca-certs" scripts reads all the certs in a certain path (including the OpenShift one via symlink), it only needs read access to those. But it generates the OpenSSL and JKS stores, and needs write permission to put them in the system location.

> Also, can the pod user run the update script without special privs?

Yes if the permission change has taken place, that's the only additional priv required.

> Will the update script fail if the link is broken (i.e. the target file doesn't exist)? 

No, silently ignored.

>  Do we need to chmod the link, to make sure the perms are correct there (I had this problem in other places, where the link was root:root, which required an explicit chown -h.

I don't think so, but in my testing I've just been launching the containers with `docker` and volume-mounting a certificate with `-v`. It's not impossible that OpenShfit does something subtly different which could expose a problem here.

Reply:
make sure you try running the container with something like `-u 10010000`

Reply:
whoops, 10010000:0

Reply:
@rcernich the test harness I wrote to check this stuff out, I published [here](https://github.com/jmtd/redhat-openjdk-18-openshift-image/tree/cloud-2689-certificates-TEST). In particular see [test.sh line 1])(https://github.com/jmtd/redhat-openjdk-18-openshift-image/blob/cloud-2689-certificates-TEST/test.sh#L1): each invocation of the test harness uses a random uid, just like openshift.

Reply:
Yes, but OpenShift also runs with group 0.

Reply:
Ah I missed the group spec! I'll adjust my harness and re-test.

Reply:
[fixed](https://github.com/jmtd/redhat-openjdk-18-openshift-image/commit/f04e2136c9088eafadbb767fae1226f428b691fa) and re-tested locally; seems fine